modular_skyrat/modules/mob_container/mob_container.dm

```diff
@@ -0,0 +1,94 @@
+#define COMSIG_MOB_CONTAINER_ENTERED "mob_container_entered"
+#define COMSIG_MOB_CONTAINER_EXITED "mob_container_left"
+#define COMSIG_MOB_ENTER_CONTAINER "mob_enter_container"
+#define COMSIG_MOB_EXIT_CONTAINER "mob_exit_container"
+
+#define COMSIG_MOBS_BREATHE_IN_CONTAINER "mob_breathe_in_container"
+
+/mob/proc/handle_internal_lifeform(mob/lifeform_inside)
+	if(SEND_SIGNAL(src, COMSIG_MOBS_BREATHE_IN_CONTAINER))
+		var/datum/gas_mixture/environment = return_air()
+		var/breath_percentage = BREATH_VOLUME / environment.return_volume()
+		return remove_air(environment.total_moles() * breath_percentage)
+
+/datum/component/mob_container
+	/// The mob we're attached to.
+	var/mob/mob_holder
+	/// The mobs we're currently holding in our container.
+	var/list/mobs_held = list()
+
+	/// The fullscreen overlay we'll add to any mob inside us. Set this to null to add no overlay.
+	var/atom/movable/screen/fullscreen_overlay = /atom/movable/screen/fullscreen/impaired
+
+	/// Whether or not mobs should be able to breathe inside of us.
+	var/container_breathable = TRUE
+
+/datum/component/mob_container/Initialize(mob/mob_holder)
+	if(!istype(mob_holder))
+		return COMPONENT_INCOMPATIBLE
+
+	src.mob_holder = WEAKREF(mob_holder)
```

Comment:
two things, don't use src. and secondly, components get added to something and the parent becomes that thing, don't have the thing it's being added to passed in on initialise